### PR TITLE
[FIX] google_map: allow specification of a domain instead of ids

### PR DESCRIPTION
**Pull request by JKE-be**
This improvement enables the provision of a key that should match the
key from the dict returned by _get_gmap_domains.
It allows to obtain the domain of res.partner for the search, instead
to have a long list of IDs. Previously, we had to provide all IDs one by
 one, resulting in a lengthy URL that led to crashes.

Simultaneously, we have enhanced the map's zoom functionality to center
the point on the map and display all markers. Previously, the map would
display the world map even if you only had markers in a small area, such
as a specific country.

After this commit:
- all partners (limited to 1000) are shown and not only the partners
  from the first page.
- demo data has been cleaned, duplicate removed, lat/lont added, ...
- iframe is now lazy loaded, to avoid extra read while not clicked on
  the map view.
- limited to 80 by default, overridable in url with limit (1000 in this
  case)

### CODE REVIEW

File: addons/website_partner/controllers/main.py
Comment (qsm-odoo) [inline addons/website_partner/controllers/main.py:9]: :warning: `website_partner` does not depend on `website_google_map`

File: addons/website_google_map/controllers/main.py
Comment (qsm-odoo) [inline addons/website_google_map/controllers/main.py]: I would put this diff in the second commit, where you actually use it. Then this first commit would only be about the map zoom functionality which deserves its own commit anyway I think.

File: addons/website_google_map/controllers/main.py
Comment (qsm-odoo) [inline addons/website_google_map/controllers/main.py:40]: Nitpicking but would not it better to just not do the search in that case? Could have:
```python
domain = None
if machin1:
    domain = ...
elif machin2:
    domain = ...

partners = PartnerSudo
if domain is not None:
    partners = PartnerSudo.search(domain + [('website_published', '=', True)])


```

File: addons/website_google_map/static/src/js/website_google_map.js
Comment (qsm-odoo) [inline addons/website_google_map/static/src/js/website_google_map.js:94]: Just so I understand (did not test, just reading the code): why not setting the right zoom at initialization?

File: addons/website_google_map/static/src/js/website_google_map.js
Comment (qsm-odoo) [inline addons/website_google_map/static/src/js/website_google_map.js:89]: Missing space. Could use an arrow function too: `n => {`

File: addons/website_google_map/static/src/js/website_google_map.js
Comment (qsm-odoo) [inline addons/website_google_map/static/src/js/website_google_map.js:88]: `const` instead of `var`

File: addons/website_google_map/controllers/main.py
Comment (qsm-odoo) [inline addons/website_google_map/controllers/main.py:38]: This is triggering the rd-security review stuff, I'll let you decide. Maybe another solution would be to extract the domain computation part in a sub-method, then instead of part of a function name, it would be a boolean like `use_XXX_domain`. And in your specific app override, instead of creating `_google_map_domain_XXX`, you would extend the sub-method that creates the domain with `if use_XXX_domain: ...`. That would be as generic as your solution.

**Review by qsm-odoo — CHANGES_REQUESTED** [review]

File: addons/website_google_map/controllers/main.py
Comment (JKE-be) [reply addons/website_google_map/controllers/main.py:40]: `(0, '=', 1)` is a falsy domain, ORM is smart and don't execute anything with it, it just return an empty browseRecord 

```py
In [1]: env['res.partner'].search([('name', '=', 'ok'), (0, '=', 1)])
Out[1]: res.partner()

In [2]: env['res.partner'].search([('name', '=', 'ok')])
2024-05-14 15:13:54,657 816919 DEBUG openerp odoo.sql_db: [50.663 ms] query: SELECT "res_partner"."id" FROM "res_partner" WHERE (("res_partner"."active" = true) AND ("res_partner"."name" = 'ok')) ORDER BY "res_partner"."complete_name" ASC , "res_partner"."id" DESC  

```

Comment (qsm-odoo) [reply addons/website_google_map/controllers/main.py:40]: I was nitpicking to avoid the `(0, '=', 1)` not increasing perf :wink: You would be the only one that does `+ [(0, '=', 1)]`, but as you wish :+1:

File: addons/website_google_map/controllers/main.py
Comment (JKE-be) [reply addons/website_google_map/controllers/main.py]: all will finish in the same commit i think ;)

**Review by JKE-be — COMMENTED** [review]
I made suggested changes

**Review by qsm-odoo — APPROVED** [review]
Thanks :+1: Did not test enough to my taste yet but as it is urgent, let's merge :+1: 

https://github.com/odoo/odoo/pull/164494#discussion_r1600250112 -> I push-forced to add the `const`, it's really a detail but :shrug: `const` means the variable is not reassigned, you can still update the object it references.

@robodoo r+

**Pull request closed without merging**
